wagtail/contrib/simple_translation/views.py

```diff
@@ -0,0 +1,153 @@
+                                    child_page.copy_for_translation(locale)
+
+                                    if child_page.numchild:
+                                        _walk(child_page)
+
+                            _walk(self.object)
+                    else:
+                        self.object.copy_for_translation(locale).save()
+
+                if len(form.cleaned_data["locales"]) == 1:
+                    locales = form.cleaned_data["locales"][0].get_display_name()
+
+                else:
+                    # Note: always plural
+                    locales = _("{} locales").format(len(form.cleaned_data["locales"]))
+
+                messages.success(self.request, self.get_success_message(locales))
+
+                return redirect(
+                    self.get_success_url() or self.get_default_success_url()
+                )
+
+        context = self.get_context_data(**kwargs)
+        return self.render_to_response(context)
+
+    def dispatch(self, request, *args, **kwargs):
+        if not request.user.has_perms(["simple_translation.submit_translation"]):
+            raise PermissionDenied
+
+        self.object = self.get_object()
+        return super().dispatch(request, *args, **kwargs)
+
+
+class SubmitPageTranslationView(SubmitTranslationView):
+    title = gettext_lazy("Translate page")
+
+    def get_subtitle(self):
+        return self.object.get_admin_display_title()
+
+    def get_object(self):
+        page = get_object_or_404(Page, id=self.kwargs["page_id"]).specific
+
+        # Can't translate the root page
+        if page.is_root():
+            raise Http404
+
+        return page
+
+    def get_default_success_url(self):
+        return reverse("wagtailadmin_explore", args=[self.get_object().get_parent().id])
+
+    def get_success_message(self, locales):
+        return _("The page '{page_title}' was successfully created {locales}").format(
+            page_title=self.object.get_admin_display_title(), locales=locales
+        )
+
+
+class SubmitSnippetTranslationView(SubmitTranslationView):
+    def get_title(self):
+        return _("Translate {model_name}").format(
+            model_name=self.object._meta.verbose_name
+        )
+
+    def get_object(self):
+        model = get_snippet_model_from_url_params(
+            self.kwargs["app_label"], self.kwargs["model_name"]
+        )
+
+        if not issubclass(model, TranslatableMixin):
+            raise Http404
+
+        return get_object_or_404(model, pk=unquote(self.kwargs["pk"]))
+
+    def get_default_success_url(self):
+        return reverse(
+            "wagtailsnippets:edit",
+            args=[
+                self.kwargs["app_label"],
+                self.kwargs["model_name"],
+                self.kwargs["pk"],
+            ],
+        )
+
+    def get_success_message(self, locales):
+        return _(
+            "The {model_name} '{object}' was successfully submitted for translation into {locales}"
```

Comment:
I think we should avoid the term "submitted" as I think this implies there's a workflow. Maybe reword this to "Successfuly created translated versions of {model_name} '{object}' in {locales}"?